cogs/tmpc.py

```diff
@@ -376,16 +383,47 @@ async def kick(self, ctx: Context, member: User):
 
            member: The member to kick
         """
-        guild: Guild = ctx.guild
-        member: Union[User, Member] = guild.get_member(member.id)
-        document = await self.check_tmpc_channel(ctx, is_mod=True)
-        await document.voice.set_permissions(member, overwrite=None)
-        await document.chat.set_permissions(member, overwrite=None)
-        if member in document.voice.members:
-            await member.move_to(None)
-        embed: Embed = Embed(title="Kick",
-                             description=f"{member.mention} was kicked.")
-        await ctx.reply(embed=embed)
+        document = await self.check_tmpc_channel(ctx.author, ctx.channel.id, is_mod=True)
+
+        msg: Message = ctx.message
+        overwrites: dict = document.voice.overwrites
+        changed = False
+        kicked = ""
+        for mention in msg.mentions:
+            mention: Member
+            if mention in document.chat.members:
+                # noinspection PyBroadException
+                try:
+                    overwrites.pop(mention)
+                    changed = True
+                    kicked += mention.mention + "\n"
+                except KeyError:
+                    pass
+
+        if changed:
+            await document.voice.edit(overwrite=overwrites)
+            await document.chat.edit(overwrite=overwrites)
```

Comment:
This is not good since the voice and text chat don't have the same permissions.
For example if the channel is not hidden, then the voice channel is visible to Studenty but the text channel is not.
Do it like before with:
```
await document.voice.set_permissions(member, overwrite=None)
await document.chat.set_permissions(member, overwrite=None)
```
This then needs to be in Line 397 to work

Reply:
set_permissions would need to many requests to the discord API.
I used a solution that should work proper too.